libs/state/src/lib/rx-state.service.ts

```diff
@@ -547,7 +587,14 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
     obsOrObsWithSideEffect: Observable<S>,
     sideEffectFn?: (arg: S) => void
   ): void {
-    const sideEffect = obsOrObsWithSideEffect.pipe(catchError(e => EMPTY))
+    const sideEffect = obsOrObsWithSideEffect.pipe(
+      catchError((e) => {
+        // used injector for compatibility with https://github.com/rx-angular/rx-angular/blob/master/libs/state/docs/usage.md#inherit
+        const errorHandler = inject(ErrorHandler, InjectFlags.Optional);
+        errorHandler?.handleError(e);
```

Comment:
Last detail, why considering the `ErrorHandler` as optional? It should always be defined.

```suggestion
        const errorHandler = inject(ErrorHandler);
        errorHandler.handleError(e);
```

Reply:
He has probably added it because it throws `inject() must be called from an injection context` since there's no injector on the instruction stack.

You recommendation with this one is correct:
```ts
constructor(private errorHandler: ErrorHandler) {}
```

Since the stable version is not released yet I think it's ok to have breaking changes. We can have a migration script for that.

Reply:
We can create a migration script, but honestly, this looks bad:

```ts
  constructor(private errorHandler: ErrorHandler) {
    super(this.errorHandler);
  }
  ```
  
I would prefer keeping the `InjectFlags.Optional` hack to hide this noise from the user.

Reply:
I've just checked that code and it seems like it's working only inside tests for now because the injector is explicitly set there. E.g. this doesn't work even with inject flags:
```ts
@Component({
  selector: 'app-root',
  template: '<button (click)="onClick()">Click me</button>',
})
export class AppComponent {
  onClick(): void {
    console.log(inject(ErrorHandler, InjectFlags.Optional));
  }
}
```

Reply:
> 
> I'm wondering if it can be moved to the constructor but it also has to be wrapped into try-catch to prevent the error if there is no current injector:
> 
> ```ts
> export class ... {
>   private errorHandler: ErrorHandler | null;
> 
>   constructor() {
>     try {
>       this.errorHandler = inject(ErrorHandler);
>     } catch {
>       this.errorHandler = null;
>     }
>   }
> ```

this sounds like the only valid solution for me since the approach with extending from RxState is pretty common.

Reply:
How about this?
```javascript
class ... {
  private errorHandler: ErrorHandler

  constructor() {
    try {
      this.errorHandler = inject(ErrorHandler)
    } catch {
      const injector = Injector.create({
        providers: [{ provide: ErrorHandler, deps: [] }]
      });
      this.errorHandler = injector.get(ErrorHandler);
    }
  }
```

Reply:
This will create `ErrorHandler`s per each RxState instance and will have nothing to do with the global error handler (the one provided by a user).

Reply:
To conclude, we should try this approach and decide:

```ts
export class ... {
  private errorHandler: ErrorHandler | null;

  constructor() {
    try {
      this.errorHandler = inject(ErrorHandler);
    } catch {
      this.errorHandler = null;
    }
  }
```